networkit/cpp/centrality/DynBetweennessOneNode.cpp

```diff
@@ -138,58 +144,66 @@ void DynBetweennessOneNode::update(GraphEvent event) {
+                                sigmax[u][v] += 1;
+                            } else {
+                                sigmax[u][v] = 0;
+                            }
                         }
+                        distances[s][y] = weightuv;
+
                     } else {
                         if (distances[s][y] > distances[s][u] + weightuv + distances[v][y]) {
                             if (s == 1 && y == 2)
                                 INFO(" GREATER s = ", s, ", y = ", y);
                             sigma[s][y] = sigma[s][u] * sigma[v][y];
                             sigmax[s][y] = sigmax[s][u] * sigma[v][y] + sigma[s][u] * sigmax[v][y];
+                            distances[s][y] = distances[s][u] + weightuv + distances[v][y];
                         } else {
                             if (s == 1 && y == 2)
                                 INFO(" EQUAL s = ", s, ", y = ", y);
                             sigma[s][y] += sigma[s][u] * sigma[v][y];
                             sigmax[s][y] += sigmax[s][u] * sigma[v][y] + sigma[s][u] * sigmax[v][y];
                         }
                     }
-                    if (s != x && y != x && sigma[s][y]) {
-                        bcx += double(sigmax[s][y]) / sigma[s][y];
-                        if (!G.isDirected()) {
-                            bcx += double(sigmax[s][y]) / sigma[s][y];
-                        }
-                    }
-                    distances[s][y] = distances[s][u] + weightuv + distances[v][y];
                     if (!G.isDirected()) {
                         distances[y][s] = distances[s][y];
                         sigma[y][s] = sigma[s][y];
                         sigmax[y][s] = sigmax[s][y];
                     }
                     source_nodes[y].push_back(s);
                 }
+                // loop over all neighbors
+                G.forNeighborsOf(y, [&](node w, edgeweight weightyw) {
+                    // I also check that y was a predecessor for w in the s.p. from v
+                    if (distances[u][w] >= weightuv + distances[v][w]
+                        && Aux::NumericTools::equal(distances[v][w], distances[v][y] + weightyw,
+                                                    0.01)) {
+                        Pred[w] = y;
+                        updateQueue(w);
+                    }
+                });
             }
-            // loop over all neighbors
-            G.forNeighborsOf(y, [&](node w, edgeweight weightyw) {
-                // I also check that y was a predecessor for w in the s.p. from v
-                if (distances[u][w] >= weightuv + distances[v][y] + weightyw
-                    && distances[v][w] == distances[v][y] + weightyw) {
-                    Pred[w] = y;
-                    updateQueue(w);
-                }
-            });
         }
+        // computing bc(x)
+        bcx = 0;
+        G.forNodePairs([&](node s, node y) {
```

Comment:
This has complexity `O(n^2)`, which we should avoid. Before the `bcx` score was computed during the loop. Seems that the logic is different now (`forNodePairs` iterates over the upper triangle of the graph instead of incrementing/decrementing twice as before).

Reply:
The `run()` method now produces correct betweenness scores for weighted graphs, but indeed, takes longer than previously. We could practically have both variants implemented and choose the one suited for the input graph (not the prettiest solution, but for the unweighted case runtimes should remain similar). 

On the other hand, our current `SSSP` variants do not store the shortest paths explicitly (rather they only store the predecessor(s) for each node). This makes computing the `bc(x)` complicated without designing a new `BFS-like` algorithm that computes `bc(x)` on-the-fly. Here, any recommendations on your behalf are welcome.